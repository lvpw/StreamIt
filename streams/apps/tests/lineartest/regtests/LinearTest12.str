/** Test combining pipelines when they don't rate match, and when they have constant vector components. **/
void->void pipeline LinearTest12 {
  add FloatSource();
  add LinearPipe();
  add LinearPipe2();
  add FloatSink();
}

void->float stateful filter FloatSource {
  float x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = x + 1;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/** Pipeline of linear filters which have constant components. **/
float->float pipeline LinearPipe {
  add FloatFilter1();
  add FloatFilter2();
}
/** pipeline of linear filters which have constant components in additon to not rate matching. **/
float->float pipeline LinearPipe2 {
  add FloatFilter3();
  add FloatFilter4();
}



/** filter that computes [[1 2][3 4][5 6][7 8]] + [9 10] **/
float->float filter FloatFilter1 {
  work pop 4 push 2 {
    float t1 = pop();
    float t2 = pop();
    float t3 = pop();
    float t4 = pop();
    push(2*t4 + 4*t3 + 6*t2 + 8*t1 + 10);
    push(1*t4 + 3*t3 + 5*t2 + 7*t1 + 9);
  }
}
/** filter computes [[1 2 3][4 5 6]] + [3 2 1] **/
float->float filter FloatFilter2 {
  work pop 2 push 3 {
    float t1 = pop();
    float t2 = pop();
    push(3*t2 + 6*t1 + 1);
    push(2*t2 + 5*t1 + 2);
    push(1*t2 + 4*t1 + 3);
  }
}

/** filter that computes [[1 2][3 4][5 6]] + [9 10] **/
float->float filter FloatFilter3 {
  work pop 3 push 2 {
    float t2 = pop();
    float t3 = pop();
    float t4 = pop();
    push(2*t4 + 4*t3 + 6*t2 + 10);
    push(1*t4 + 3*t3 + 5*t2 + 9);
  }
}
/** Filter computes [[1 2 1][2 1 2][3 3 3]] + [2 3 1] **/
float->float filter FloatFilter4 {
  work pop 3 push 3 {
    float t1 = pop();
    float t2 = pop();
    float t3 = pop();
    push(t3*1 + t2*2 + t1*3);
    push(t3*2 + t2*1 + t1*3);
    push(t3*1 + t2*2 + t1*3);
  }
}
