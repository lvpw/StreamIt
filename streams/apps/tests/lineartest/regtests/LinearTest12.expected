FloatFilter1_1.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]
 [5.0+0.0i 6.0+0.0i]
 [7.0+0.0i 8.0+0.0i]]
FloatFilter1_1.vector
[[9.0+0.0i 10.0+0.0i]]
FloatFilter2_1.matrix
[[1.0+0.0i 2.0+0.0i 3.0+0.0i]
 [4.0+0.0i 5.0+0.0i 6.0+0.0i]]
FloatFilter2_1.vector
[[3.0+0.0i 2.0+0.0i 1.0+0.0i]]
FloatFilter3_1.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]
 [5.0+0.0i 6.0+0.0i]]
FloatFilter3_1.vector
[[9.0+0.0i 10.0+0.0i]]
FloatFilter4_1.matrix
[[1.0+0.0i 2.0+0.0i 1.0+0.0i]
 [2.0+0.0i 1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 3.0+0.0i 3.0+0.0i]]
FloatFilter4_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearPipe2_1.matrix
[[5.0+0.0i 4.0+0.0i 5.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [11.0+0.0i 10.0+0.0i 11.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [17.0+0.0i 16.0+0.0i 17.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [3.0+0.0i 3.0+0.0i 3.0+0.0i 2.0+0.0i 4.0+0.0i 2.0+0.0i]
 [9.0+0.0i 9.0+0.0i 9.0+0.0i 4.0+0.0i 8.0+0.0i 4.0+0.0i]
 [15.0+0.0i 15.0+0.0i 15.0+0.0i 6.0+0.0i 12.0+0.0i 6.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 8.0+0.0i 7.0+0.0i 8.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 18.0+0.0i 15.0+0.0i 18.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 28.0+0.0i 23.0+0.0i 28.0+0.0i]]
LinearPipe2_1.vector
[[56.0+0.0i 55.0+0.0i 56.0+0.0i 58.0+0.0i 59.0+0.0i 58.0+0.0i]]
LinearPipe_1.matrix
[[9.0+0.0i 12.0+0.0i 15.0+0.0i]
 [19.0+0.0i 26.0+0.0i 33.0+0.0i]
 [29.0+0.0i 40.0+0.0i 51.0+0.0i]
 [39.0+0.0i 54.0+0.0i 69.0+0.0i]]
LinearPipe_1.vector
[[52.0+0.0i 70.0+0.0i 88.0+0.0i]]
LinearWrapper_1.matrix
[[432.0+0.0i 396.0+0.0i 432.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [942.0+0.0i 864.0+0.0i 942.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [1452.0+0.0i 1332.0+0.0i 1452.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [1962.0+0.0i 1800.0+0.0i 1962.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [360.0+0.0i 360.0+0.0i 360.0+0.0i 156.0+0.0i 312.0+0.0i 156.0+0.0i]
 [786.0+0.0i 786.0+0.0i 786.0+0.0i 340.0+0.0i 680.0+0.0i 340.0+0.0i]
 [1212.0+0.0i 1212.0+0.0i 1212.0+0.0i 524.0+0.0i 1048.0+0.0i 524.0+0.0i]
 [1638.0+0.0i 1638.0+0.0i 1638.0+0.0i 708.0+0.0i 1416.0+0.0i 708.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 708.0+0.0i 588.0+0.0i 708.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 1544.0+0.0i 1282.0+0.0i 1544.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 2380.0+0.0i 1976.0+0.0i 2380.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 3216.0+0.0i 2670.0+0.0i 3216.0+0.0i]]
LinearWrapper_1.vector
[[4688.0+0.0i 4477.0+0.0i 4688.0+0.0i 5110.0+0.0i 5321.0+0.0i 5110.0+0.0i]]
